testTypeHttp.py

```diff
@@ -0,0 +1,98 @@
+import json
+import shlex
+
+from dataclasses import dataclass
+
+import common
+import perf
+import tftbase
+
+from logger import logger
+from perf import PerfClient
+from perf import PerfServer
+from task import TaskOperation
+from testSettings import TestSettings
+from testType import TestTypeHandler
+from tftbase import BaseOutput
+from tftbase import IperfOutput
+from tftbase import TestType
+
+
+@dataclass(frozen=True)
+class TestTypeHandlerHttp(TestTypeHandler):
+    def __init__(self) -> None:
+        super().__init__(TestType.HTTP)
+
+    def _create_server_client(self, ts: TestSettings) -> tuple[PerfServer, PerfClient]:
+        s = HttpServer(ts=ts)
+        c = HttpClient(ts=ts, server=s)
+        return (s, c)
+
+
+test_type_handler_http = TestTypeHandlerHttp()
+
+
+class HttpServer(perf.PerfServer):
+    def cmd_line_args(self) -> list[str]:
+        return [
+            "-m",
+            "http.server",
+            "-d",
+            "/etc/ocp-traffic-flow-tests",
+            f"{self.port}",
+        ]
+
+    def get_template_args(self) -> dict[str, str]:
+
+        extra_args: dict[str, str] = {}
+        if self.exec_persistent:
+            extra_args["command"] = "python3"
+            extra_args["args"] = json.dumps(self.cmd_line_args())
+
+        return {
+            **super().get_template_args(),
+            **extra_args,
+        }
+
+    def _create_setup_operation_get_thread_action_cmd(self) -> str:
+        return f"python3 {shlex.join(self.cmd_line_args())}"
+
+    def _create_setup_operation_get_cancel_action_cmd(self) -> str:
+        return "killall python3"
+
+
+class HttpClient(perf.PerfClient):
+    def _create_task_operation(self) -> TaskOperation:
+        server_ip = self.get_target_ip()
+        cmd = f"exec {self.pod_name} -- curl --fail -s http://{server_ip}:{self.port}/data"
+
+        def _thread_action() -> BaseOutput:
+            self.ts.clmo_barrier.wait()
+            r = self.run_oc(cmd)
+            self.ts.event_client_finished.set()
+
+            return IperfOutput(
```

Comment:
"IperfOutput" doesn't look right here.

Reply:
I think `PerfClient` and `PerfServer` should be renamed to `TaskClient` and `TaskServer`.
And `perf.py` (the file that contains those two classes) should just merge with `task.py`.
And `IperfOutput` should be renamed to `FlowTestOutput`.

That renaming will happen in a later patch (that is not yet written). But for now, the `Evaluator` operates on `flow_test IperfOutput`. Despite the naming, this is not Iperf-specific...

Reply:
Ok thanks! Glad you noticed this too